receiver/dotnetdiagnosticsreceiver/metrics/sender.go

```diff
@@ -0,0 +1,47 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver/dotnet"
+)
+
+// Sender wraps a consumer.MetricsConsumer, and has a Send method which
+// conforms to dotnet.MetricsConsumer so it can be passed into a Parser.
+type Sender struct {
+	next      consumer.MetricsConsumer
+	startTime time.Time
+	logger    *zap.Logger
+}
+
+func NewSender(next consumer.MetricsConsumer, startTime time.Time, logger *zap.Logger) *Sender {
+	return &Sender{next: next, startTime: startTime, logger: logger}
+}
+
+// Send accepts a slice of dotnet.Metrics, converts them to pdata.Metrics, and
+// sends them to the next pdata consumer. Conforms to dotnet.MetricsConsumer.
+func (s Sender) Send(rawMetrics []dotnet.Metric) {
+	pdm := rawMetricsToPdata(rawMetrics, s.startTime, time.Now())
+	err := s.next.ConsumeMetrics(context.Background(), pdm)
+	if err != nil {
+		s.logger.Error(err.Error())
```

Comment:
I think conventionally there's desired `obsreport` usage to help provide introspection on how this fairs.  Could be introduced in a later PR though.

Reply:
Will look into this for a subsequent PR, thanks.